### DISCUSSION TITLE
h5netcdf - an alternative to `separate_complex_values`, `merge_complex_values`

### DISCUSSION THREAD

**Discussion by ckesanapalli** [capytaine/capytaine#225] [Ideas]
An alternative for faster reading of the netcdf files. 

Using Capytaine's IO functions for reading and writing

```python
import xarray as xr
from capytaine.io.xarray import separate_complex_values, merge_complex_values

# Reading a xarray dataset
ds = merge_complex_values(xr.open_dataset("data.nc"))
# 00.423 sec
# 

# Writing a xarray dataset
separate_complex_values(ds).to_netcdf("data.nc")
# 12.70 sec
```

Using Xarray's default for reading and writing
```python
import xarray as xr

# Reading a xarray dataset
ds = xr.open_dataset("data.nc", engine="h5netcdf")
# 00.15 sec

# Writing a xarray dataset
ds.to_netcdf("data.nc", engine="h5netcdf", invalid_netcdf=True)
# 12.92 sec
#Warning
#The file will thus be not conforming to NetCDF-4 standard and might 
#not be readable by other netcdf tools. Consider using a different extension.
#  self.flush()
```

For large netcdf files, the xarray default read significantly faster than with `merge_complex_values` whereas when writing it is slightly slower than `separate_complex_values`. 

Also note that, using `invalid_netcdf=True` is required resulting in non-compatibility with other NETCDF reader. 

A viable alternative if a user want a quick IO.

**Comment by mancellin** [comment]
The purpose of `separate_complex_values` and `merge_complex_values` is only to have a file in a standard format that could be read somewhere else.
Feel free to ignore them if compatibility is not an issue for you!
